kalite/knowledgemap/templates/knowledgemap/knowledgemap.html

```diff
@@ -0,0 +1,616 @@
+        var TOPIC_DATA_URLS = {
+            Domains: '{{ data_url }}/map_domains.json',
+            Subjects: '{{ data_url }}/map_subjects.json',
+            Topics: '{{ data_url }}/topics.json',
+            Tutorials: '{{ data_url }}/map_tutorials.json',
+            Exercises: '{{ data_url }}/exercises.json'
+        }
+        var ALL_TOPICS_URL = '{{ topics_url }}';
+    </script>
+{% endblock headjs %}
+
+{% block content %}
+<script src="http://d3js.org/d3.v3.min.js"></script>
+
+<script>
+window.statusModel.loaded.then(function() {
+
+  //add map explanation
+  var graph_explain_img=document.createElement("img");
+  graph_explain_img.style.position="absolute";
+  graph_explain_img.style.top="70px";
+  graph_explain_img.style.left="10px";
+  graph_explain_img.style.width="100px";
+  graph_explain_img.setAttribute('src', "{% static 'images/distributed/map_explain_img.png' %}");
+  document.body.appendChild(graph_explain_img);
+
+  var user_exercises_json;
+  var url = setGetParamDict("/api/knowledgemap/KnowledgeMapExerciselog/", {
+    user: statusModel.get("user_id"), 
+    completion_timestamp__gte: '2011-10-20',   //because the topic.json contain repetitive object under same parent, which cause problem for displaying knowledge map
+    completion_timestamp__lte: '2017-12-22'     //use this time filter to avoid encounter repetitive objects
+  }); 
+
+  doRequest(url)
+          .success(function(user_json) {
+            user_exercises_json = user_json;
+
+  //create the hashtatble 
+  var user_hashtable = {};
+  for (var key in user_exercises_json["objects"]) {
+    user_hashtable[user_exercises_json["objects"][key].exercise_id] = user_exercises_json["objects"][key];
+  }
+
+  // color-order: math_blue/sifi_scarlet/econ_orange/artist_red/cyber_green/test_purple/partner_emerald/college_pink
+  var color_scheme_array = ["#1c758a","#94424f","#c78d46","#cf5044","#699c52","#644172","#49a88f","#e89cb9"];
+
+  var subject_list = [];
+
+  var width = window.innerWidth,
+    height = window.innerHeight,
+    root;
+
+  var force = d3.layout.force()
+    // // .theta(.0001)
+    // .alpha(0.1)
+    .size([width, height]);
+    // .on("tick", tick);
+
+  $( ".kalite-navbar" ).after(function() {
+    return "<kmap></kmap>";
+  });
+
+  var svg = d3.select("kmap").append("svg")
+    .attr("width", "100%")
+    .attr("height", window.innerHeight - 180)
+    .call(d3.behavior.zoom().scaleExtent([0.01, 30]).on("zoom", zoomHandler))
+    .append("g");
+
+  var label_translate;
+  var label_scale;
+  function zoomHandler() {
+    label_translate = d3.event.translate;
+    label_scale = d3.event.scale;
+    svg.attr("transform", "translate(" + label_translate + ")scale(" + label_scale + ")");
+  }
+
+  var link = svg.selectAll(".link"),
+      node = svg.selectAll(".node");
+
+  var initialized =false;
+
+  var subj_btn_top = 70;
+  var subj_btn_left = 30;
+
+  d3.json(sprintf(TOPIC_DATA_URLS.Topics, {channel_name: "khan"}), function(error, json) {
```

Comment:
This fetch could probably happen earlier, it would not even have to wait until the statusModel is loaded.